app/Services/SubscribeService.php

```diff
@@ -4,37 +4,43 @@
 
 namespace App\Services;
 
+use App\Dto\StoreAdvertDto;
 use App\Jobs\SubscribeJob;
 use App\Models\Advert;
 use App\Models\User;
 use App\Notifications\AdvertMissingNotification;
-use App\Notifications\SubscribeNotification;
-use Illuminate\Support\Facades\Notification;
 
 class SubscribeService
 {
+    private ParserService $parser;
+    private NotifyService $notifier;
 
-    private array $advertData;
-    private int $createdAdvertId;
-    private ParserService $service;
-
-    public function __construct(ParserService $service)
+    public function __construct(ParserService $parser, NotifyService $notifier)
     {
-        $this->service = $service;
+        $this->parser = $parser;
+        $this->notifier  = $notifier;
     }
 
     /**
      * Dispatch job
      * @param User $user
-     * @param array $data
+     * @param StoreAdvertDto $dto
      * @return string
      */
-    public function subscribe(User $user, array $data): string
+    public function subscribe(User $user, StoreAdvertDto $dto): string
     {
-        $data['email'] = $data['email'] ?? null;
-        if ($user->adverts()->where('url', $data['url'])->get()->count() === 0) {
-            $data['advertId'] = $this->createAdvert($data['url']);
-            SubscribeJob::dispatch($user, $data);
+        $email = $dto->email ?? null;
```

Comment:
Consider to add getter to DTO. Like $dto->getEmail() and return email or null from it